packages/bpk-component-chip-group/src/BpkChipGroupSingleSelect.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+import BpkChipGroup, { type ChipItem, type SingleSelectChipItem, type ChipGroupProps } from './BpkChipGroup';
+
+type CommonSingleSelectProps = {
+  chips: SingleSelectChipItem[];
+  onItemClick?: (item: SingleSelectChipItem, selected: boolean, index: number) => void,
+} & ChipGroupProps;
+
+export type SingleSelectProps = {
+  selectedIndex?: number;
+} & CommonSingleSelectProps;
+
+const BpkChipGroupSingleSelect = ({ chips, onItemClick, selectedIndex, ...rest }: SingleSelectProps) => {
+  const chipsWithSelection = chips.map((chip, index) => chip && ({
+    ...chip,
+    selected: index === selectedIndex,
+    onClick: (selected: boolean, clickedIndex: number) => {
+      if (onItemClick) {
+        onItemClick(chip, selected, clickedIndex);
+      }
+    },
+  }));
+
+  return (
+    <BpkChipGroup chips={chipsWithSelection} ariaMultiselectable={false} {...rest} />
+  );
+};
+
+export type SingleSelectStateProps = {
+  initiallySelectedIndex?: number;
+} & CommonSingleSelectProps
+
+export const BpkChipGroupSingleSelectState = ({ initiallySelectedIndex = -1, onItemClick, ...rest }: SingleSelectStateProps) => {
```

Comment:
Same as previous comment

> I wonder if there is something more clever we can do here instead of exposing two different types of chips just a difference of having state.
>
> What is the difference of needing to have a stateless and stateful chip? I think this should either be put to the consumer (like all our other components as Backpack shouldn't know about state) or built in mechanisms to detect to use state

Reply:
It seemed like a nice utility to provide, especially since we had to implement it to show examples anyway. Alternatively this could be a HOC like some other components do. e.g. `withMultiSelectState(BpkMultiSelectChipGroup)` however I originally decided against this as I think we would still need 2 separate functions for MultiSelect and SingleSelect - so it only adds complexity to the consumer doing the importing and the setup before using the component